integration-test/src/main/java/com/sequenceiq/it/cloudbreak/actor/CloudbreakUserCache.java

```diff
@@ -1,66 +1,152 @@
+
+    protected CommonCloudProperties commonCloudProperties() {
+        return commonCloudProperties;
     }
 
-    public static CloudbreakUserCache getInstance() {
-        CloudbreakUserCache result = instance;
-        if (result == null) {
-            synchronized (mutex) {
-                result = instance;
-                if (result == null) {
-                    result = new CloudbreakUserCache();
-                    instance = result;
-                }
-            }
-        }
-        return result;
+    public CloudbreakUser getByDisplayName(String name) {
+        return getByDisplayName(name, commonCloudProperties.getUser().getRealUms().getEnvironmentKey(),
+                commonCloudProperties.getUser().getRealUms().getAccountKey());
     }
 
-    public CloudbreakUser getByName(String name) {
-        if (usersByAccount == null) {
-            initUsers();
+    public CloudbreakUser getByDisplayName(String name, String environmentKey, String accountKey) {
+        setRealUmsUserEnvironment(environmentKey);
+        setRealUmsUserAccount(accountKey);
+
+        if (usersByAccount == null || !initializedKeys.containsKey(environmentKey) || !initializedKeys.containsValue(accountKey)) {
+            initUsers(environmentKey, accountKey);
         }
-        return usersByAccount.values().stream().flatMap(Collection::stream)
-                .filter(u -> u.getDisplayName().equals(name)).findFirst().get();
+        CloudbreakUser user = usersByAccount.values().stream().flatMap(Collection::stream)
+                .filter(u -> u.getDisplayName().equals(name)).findFirst()
+                .orElseThrow(() -> new TestFailException(String.format("There is no real UMS user with name: %s in account: %s", name, accountKey)));
+        LOGGER.info(" Real UMS user has been found in repository:: \nDisplay name: {} \nCrn: {} \nAccess key: {} \nSecret key: {} \nAdmin: {} ",
+                user.getDisplayName(), user.getCrn(), user.getAccessKey(), user.getSecretKey(), user.getAdmin());
+        return user;
+    }
+
+    public void setUsersByAccount(Map<String, List<CloudbreakUser>> users) {
+        this.usersByAccount = users;
+    }
+
+    public Map<String, List<CloudbreakUser>> getUsersByAccount() {
+        return usersByAccount;
     }
 
-    public void initUsers() {
+    public String getRealUmsUserAccount() {
+        return realUmsUserAccount;
+    }
+
+    public void setRealUmsUserAccount(String realUmsUserAccountKey) {
+        this.realUmsUserAccount = realUmsUserAccountKey;
+    }
+
+    public String getRealUmsUserEnvironment() {
+        return realUmsUserEnvironment;
+    }
+
+    public void setRealUmsUserEnvironment(String realUmsUserEnvironmentKey) {
+        this.realUmsUserEnvironment = realUmsUserEnvironmentKey;
+    }
+
+    public void setInitializedKeys(Map<String, String> keys) {
+        this.initializedKeys = keys;
+    }
+
+    public void initUsers(String environmentKey, String accountKey) {
```

Comment:
What is the difference between environmentKey and accountKey?

Reply:
Would it be feasible to make this to an enum? That way if people see the possible values, like dev,int,stage,prod might get the idea on what this parameter means.

Or the other idea is to change it to deployment?

Reply:
The new `resources/ums-users/api-credentials.json` structure: 
```
{
  "dev": {
    "default": [...],
    "legacy": [...]
  },
  "int": {},
  "stage": {}
}
```
here the `environmentKey` can be `dev, int, stage` and `accountKey` can be `default, legacy`.

Reply:
Yes, we can change `environmentKey` to ENUM or rename to `deploymentKey`.